### DISCUSSION TITLE
Auth with Email Code?

### DISCUSSION THREAD

**Discussion by osseonews** [nhost/nhost#2642] [Q&A]
I noticed that with Auth you have Magic Link option and you offer a one-time otp code with SMS, but what about a code with email? The problem nowadays, is that nobody wants to click on any link in emails and in many companies these links will be disabled. At the same time, SMS is not ideal because it requires another third-party.  Another, more popular option, nowadays, is to send a code by email to allow login (functions the same as a mobile number, but the code is sent by email). How do we do this with Nhost? I would mention that every Auth provider offers this now. I imagine this is easy to ipmlement, as the logic seems similar to an SMS, except that the code is sent by email. Has anything been written to do this?

**Comment by dbarrosop** [comment]
That is supported already using [custom email templates](https://docs.nhost.io/guides/auth/email-templates).

The idea is:

1. Modify the template to send, instead of a link, the ticket (the code you are referring to). You can check the variables we provide you [here](https://docs.nhost.io/guides/auth/email-templates#template-variables).
2. Build a page where the user can enter the code
3. Once the user enters the code that same page can perform the fetch.

If you have further questions let us know.

**Reply by osseonews** [reply]
Yeah, but the "ticket" is just some random long code that nobody can possibly type in without a cut and paste and is unintelligible to the user. So it's not a flow that is familiar. What we are looking for is the OTP like experience you get with the SMS, but send the OTP to email instead of SMS. So it's the same exact thing, but the auth would happen by passing the email and OTP vs a phone number and OTP. This is kind of how most auth works now. You get a choice of sending OTP to email or SMS.

**Reply by dbarrosop** [reply]
> Yeah, but the "ticket" is just some random long code that nobody can possibly type in without a cut and paste and is unintelligible to the user.

What is the concern with this though? Just curious about the concern/use case. From a security point of view it is much better than just 4-6 digits and a copy&paste seems like a more convenient alternative to writing by hand.

**Comment by osseonews** [comment]
"copy&paste seems like a more convenient alternative to writing by hand.". Not for 99% of people. From a user perspective, you want to write 6 numbers, as you can remember that by heart, vs doing cut and paste. That's why every single auth does it this way now. Nobody sends you some unintellible code to enter. It's just a series of numbers, like you do for SMS. I don't want to cut and paste anything. Nor does anyone else I know. If I get an email, I get a pop screen on my computer or phone and I see the number, remember it, and type it in. No need to do some cut and paste. Anyway, there are many scenarios where you can't even cut and paste without difficulty or you need to give the code to another trusted user for some reason (e.g. when my wife screams at me from downstairs, what's the 6 digit code they just sent you?).

**Reply by dbarrosop** [reply]
Ok, thanks for the explanation. Feel free to open an issue with the feature request in this same repo.
